From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Tromey <tom@tromey.com>
Date: Wed, 10 Jun 2020 20:38:27 -0600
Subject: [PATCH] Remove "throw" specifications

C++ throw specifications were deprecated in C++11.
This patch removes them from the library.
---
 ChangeLog                 | 5 +++++
 lib/srchilite/fileutil.cc | 2 +-
 lib/srchilite/fileutil.h  | 2 +-
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index fcc12ff..c194301 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2020-06-10  Tom Tromey  <tom@tromey.com>
+
+	* lib/srchilite/fileutil.cc (readFile): Remove "throw".
+	* lib/srchilite/fileutil.h (readFile): Remove "throw".
+
 2020-06-10  Vedran Vidovic  <vvidovic>
 
 	* tests/lang.list: Update.
diff --git a/lib/srchilite/fileutil.cc b/lib/srchilite/fileutil.cc
index 59a6d64..963178c 100644
--- a/lib/srchilite/fileutil.cc
+++ b/lib/srchilite/fileutil.cc
@@ -48,7 +48,7 @@ void set_file_util_verbose(bool b) {
 // FIXME avoid using a global variable
 std::string start_path;
 
-string readFile(const string &fileName) throw (IOException) {
+string readFile(const string &fileName) {
     ifstream file(fileName.c_str());
 
     if (!file.is_open()) {
diff --git a/lib/srchilite/fileutil.h b/lib/srchilite/fileutil.h
index 7335a9b..042eb56 100644
--- a/lib/srchilite/fileutil.h
+++ b/lib/srchilite/fileutil.h
@@ -27,7 +27,7 @@ extern std::string start_path;
  * @return the contents of the file
  * @throw IOException
  */
-string readFile(const string &fileName) throw (IOException);
+string readFile(const string &fileName);
 
 //char *read_file(const string &fileName);
 
